File: Go/rock.go
package main

import "github.com/hajimehoshi/ebiten/v2"

type Rock struct {
	position *Vector2
	velocity *Vector2
	gravity  *Vector2
}

func (rock *Rock) Construct(Position Vector2, Velocity Vector2) {
	rock.position = &Position
	rock.velocity = &Velocity
	rock.gravity = &Vector2{0, 0.98}
}

func (rock *Rock) Update(water *Water) {
	if rock.position.y > water.GetHeight(rock.position.x) {
		rock.velocity.x *= 0.84
		rock.velocity.y *= 0.84
	}
	rock.position.x += rock.velocity.x
	rock.position.y += rock.velocity.y
	rock.velocity.x += rock.gravity.x
	rock.velocity.y += rock.gravity.y
}

func (rock *Rock) Draw(screen *ebiten.Image, image *ebiten.Image) {
	rockPos := &ebiten.DrawImageOptions{}
	rockPos.GeoM.Translate(rock.position.x, rock.position.y)
	screen.DrawImage(image, rockPos)
}

File: Java/core/src/com/water/gdx/Rock.java
package com.water.gdx;

import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.math.Vector2;

public class Rock {

    public Vector2 Position, Velocity;
    static final Vector2 Gravity = new Vector2(0, 0.3f);

    public Rock(Vector2 Position, Vector2 Velocity) {
        this.Position = Position;
        this.Velocity = Velocity;
    }

    public void Update(Water water) {
        if (Position.y > water.GetHeight(Position.x)) {
            Velocity.x *= 0.84f;
            Velocity.y *= 0.84f;
        }
        Position.x += Velocity.x;
        Position.y += Velocity.y;
        Velocity.x += Gravity.x;
        Velocity.y += Gravity.y;
    }

    public void Draw(SpriteBatch spriteBatch, Texture texture) {
        spriteBatch.draw(texture, Position.x, Position.y, texture.getWidth(), texture.getHeight(), 0, 0, texture.getWidth(),
                texture.getHeight(), false, true);
    }

}
